code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1899,10 +1899,22 @@ mob/living/carbon/human/isincrit()
 
 /mob/living/carbon/human/Hear(var/datum/speech/speech, var/rendered_speech="")
 	..()
-	if(!mind.faith || speech.frequency || speech.speaker == src || !ishuman(speech.speaker) || length(speech.message) < 20)
-		return
+	if(speech.frequency || speech.speaker == src || !ishuman(speech.speaker))
+		return //First, eliminate radio chatter, speech from us, or speech from nonhumans
+	var/mob/living/carbon/human/H = speech.speaker
+	var/list/diseases = H.virus2
+	if(istype(diseases) && diseases.len)
+		//do not use check_contact_sterility because other things cover ears, like helmets
+		if(ears && prob(ears.sterility))
+			return //If wearing sterile earpiece, block the meme
+		for(var/ID in diseases)
+			var/datum/disease2/disease/V = diseases[ID]
+			if(V.spread & SPREAD_MEMETIC)
+				infect_disease2(V, notes="(Memed, from [speech.speaker])")
+
+	if(!mind.faith || length(speech.message) < 20)
+		return //If we aren't religious or hearing a long message, don't check further
```

Comment:
why ignore long messages?

Reply:
That isn't related to disease code, it's an older feature. You're not ignoring long messages, you're ignoring short messages ("if we aren't... hearing a long message")

That code is related to the feature where a religious leader can talk to people to calm them down, it's not new.

Reply:
yeah i was wondering about the religious thingy, thanks for explaining